### PR TITLE
Vector of shared ptrs

### PR DESCRIPTION
**Pull request by ms10335**

### CODE REVIEW

**Comment by ziobron** [comment]
✅ CI Passed - Task: vector-of-shared-ptrs
🏅 Granted 5 XP for @ms10335

**Comment by ziobron** [comment]
⏱ 3 XP extra for @ms10335 for on-time delivery

File: homework/vector-of-shared-ptrs/vectorFunctions.cpp [hunk 1]
```diff
@@ -0,0 +1,38 @@
+#include "vectorFunctions.hpp"
+#include <iostream>
+#include <memory>
+#include <vector>
+
+std::vector<std::shared_ptr<int>> generate(int count) {
+    std::vector<std::shared_ptr<int>> result;
```
Comment (nauka-programowania-MA) [inline homework/vector-of-shared-ptrs/vectorFunctions.cpp:7]: Reserve memory for vector.

File: homework/vector-of-shared-ptrs/vectorFunctions.cpp [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    for (int i = 0; i < count; ++i) {
+        result.push_back(std::make_shared<int>(i));
+    }
+    return result;
+}
+void print(const std::vector<std::shared_ptr<int>>& result) {
+    for (const auto& a : result) {
+        std::cout << *a << " ";
```
Comment (nauka-programowania-MA) [inline homework/vector-of-shared-ptrs/vectorFunctions.cpp:15]: Ptr could be nullptr

File: homework/vector-of-shared-ptrs/vectorFunctions.cpp [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    }
+    std::cout << '\n';
+}
+void add10(std::vector<std::shared_ptr<int>>& result) {
+    for (auto& i : result) {
+        if( i){
+            *i = (*i) + 10;
```
Comment (nauka-programowania-MA) [inline homework/vector-of-shared-ptrs/vectorFunctions.cpp:22]: Ptr could be nullptr

File: homework/vector-of-shared-ptrs/vectorFunctions.cpp [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+}
+void print(const std::vector<std::shared_ptr<int>>& result) {
+    for (const auto& a : result) {
+        std::cout << *a << " ";
+    }
+    std::cout << '\n';
+}
+void add10(std::vector<std::shared_ptr<int>>& result) {
+    for (auto& i : result) {
+        if( i){
```
Comment (nauka-programowania-MA) [inline homework/vector-of-shared-ptrs/vectorFunctions.cpp:21]: remove space

File: homework/vector-of-shared-ptrs/vectorFunctions.cpp [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+            *i = (*i) + 10;
+        }
+        
+    }
+}
+void sub10(int const* ptr) {
+    if (ptr) {
+        int* result = const_cast<int*>(ptr);
```
Comment (nauka-programowania-MA) [inline homework/vector-of-shared-ptrs/vectorFunctions.cpp:29]: Wow never do that!

File: homework/vector-of-shared-ptrs/vectorFunctions.cpp [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+        *result -= 10;
+    }
+}
+void sub10(std::vector<std::shared_ptr<int>>& result) {
+    for(auto& i : result)
+    {
+        sub10(i.get());
```
Comment (nauka-programowania-MA) [inline homework/vector-of-shared-ptrs/vectorFunctions.cpp:36]: Ptr could be nullptr

**Review by nauka-programowania-MA — APPROVED** [review]

**Pull request closed without merging**